tests/test_rocm_sanity.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+import subprocess
+import re
+from pathlib import Path
+
+THIS_DIR = Path(__file__).parent
+
+
+def run_command_with_search(command, to_search):
+    process = subprocess.run(command, capture_output=True, check=False)
+    if process.returncode != 0:
+        pytest.fail(f"The command {command} failed. Failing this test...")
+    return re.search(to_search, str(process.stdout))
+
+
+class TestROCmSanity:
+    def test_rocminfo(self):
+        assert run_command_with_search(["rocminfo"], r"Device\s*Type:\s*GPU")
+
+        assert run_command_with_search(["rocminfo"], r"Name:\s*gfx")
+
+        assert run_command_with_search(["rocminfo"], r"Vendor\s*Name:\s*AMD")
+
+    def test_clinfo(self):
+        assert run_command_with_search(
+            ["clinfo"], r"Device(\s|\\t)*Type:(\s|\\t)*CL_DEVICE_TYPE_GPU"
+        )
+
+        assert run_command_with_search(["clinfo"], r"Name:(\s|\\t)*gfx")
+
+        assert run_command_with_search(
+            ["clinfo"], r"Vendor:(\s|\\t)*Advanced Micro Devices, Inc."
+        )
+
+    def test_hip_printf(self):
+        # Compiling .cpp file using hipcc
+        run_command_with_search(
+            [
+                "hipcc",
+                f"{str(THIS_DIR)}/hip_printf.cpp",
+                "-o",
+                f"{str(THIS_DIR)}/hip_printf",
+            ],
+            "",
```

Comment:
Empty search indicates that the "with_search" abstraction isn't a good fit here. Add a helper function to run a command instead?